central/node/globaldatastore/datastore_sac_test.go

```diff
@@ -0,0 +1,471 @@
+	clusterID := testconsts.Cluster2
+
+	cases := getWriteSACMultiNodeTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+
+			results, err := s.globalDatastore.GetAllClusterNodeStores(ctx, true)
+			s.NoError(err)
+
+			fetchedClusterIDs := make([]string, 0, len(results))
+			for fetchedClusterID := range results {
+				fetchedClusterIDs = append(fetchedClusterIDs, fetchedClusterID)
+			}
+			s.ElementsMatch(c.ExpectedClusterIds, fetchedClusterIDs)
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetClusterNodeStore() {
+	clusterID := testconsts.Cluster2
+
+	cases := testutils.GenericClusterSACGetTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			datastore, err := s.globalDatastore.GetClusterNodeStore(ctx, clusterID, false)
+			if c.ExpectedFound {
+				s.NoError(err)
+				s.Require().NotNil(datastore)
+			} else {
+				s.ErrorIs(err, sac.ErrResourceAccessDenied)
+				s.Nil(datastore)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetClusterNodeStoreWriteAccess() {
+	clusterID := testconsts.Cluster2
+
+	cases := testutils.GenericClusterSACWriteTestCases(context.Background(), s.T(), "write", clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			datastore, err := s.globalDatastore.GetClusterNodeStore(ctx, clusterID, true)
+			if c.ExpectError {
+				s.ErrorIs(err, sac.ErrResourceAccessDenied)
+				s.Nil(datastore)
+			} else {
+				s.NoError(err)
+				s.Require().NotNil(datastore)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestRemoveClusterNodeStores() {
+	cases := testutils.GenericGlobalSACDeleteTestCases(s.T())
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := s.testContexts[c.ScopeKey]
+
+			err := s.globalDatastore.RemoveClusterNodeStores(ctx, s.testNodeIDs[testconsts.Cluster1][1])
+			s.waitForIndexing()
+
+			if c.ExpectError {
+				s.ErrorIs(err, c.ExpectedError)
+			} else {
+				s.NoError(err)
+			}
+
+			err = s.globalDatastore.RemoveClusterNodeStores(ctx, s.testNodeIDs[testconsts.Cluster2][0], s.testNodeIDs[testconsts.Cluster2][1])
+			s.waitForIndexing()
+
+			if c.ExpectError {
+				s.ErrorIs(err, c.ExpectedError)
+			} else {
+				s.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestSearchResults() {
+	clusterID := testconsts.Cluster2
+
+	cases := getReadSACMultiNodeTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			results, err := s.globalDatastore.SearchResults(ctx, nil)
+			s.NoError(err)
+
+			fetchedNodeIDs := make([]string, 0, len(results))
+			for _, result := range results {
+				fetchedNodeIDs = append(fetchedNodeIDs, result.Id)
```

Comment:
nit: I think it would be better to append `result.GetId()`
Also applies in `SearchRawNodes`.